actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -90,18 +100,18 @@ class ActionCable.Connection
     open: ->
       ActionCable.log("WebSocket onopen event")
       @disconnected = false
```

Comment:
Re. subscription requests before welcome: we [buffer incoming messages and process the buffer after welcome](https://github.com/rails/rails/blob/master/actioncable/lib/action_cable/connection/base.rb#L168-L170), so sending early subscription requests should work as expected.